src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
+
+export const APIResponse = {
+  create: (
+    isSuccess: number,
+    statusCode: number,
+    message?: string,
+    data?: any,
+    error?: any,
+  ): ControllerResponse => {
+    const response: ControllerResponse = { isSuccess, statusCode, message };
+    if (error) {
+      response.error = error;
+    } else if (!error && data) {
+      response.data = data;
+    } else {
+      response.data = {};
+    }
+
+    return response;
```

Comment:
PullProofAI

The `APIResponse` object can be refactored to use a more functional approach by removing the object literal and directly returning the response object. This can simplify the code and make it more concise.
```typescript
export const APIResponse = {
  create: (
    isSuccess: number,
    statusCode: number,
    message?: string,
    data?: any,
    error?: any,
  ): ControllerResponse => {
    if (error) {
      return { isSuccess, statusCode, message, error };
    } else {
      return { isSuccess, statusCode, message, data: data || {} };
    }
  },
};
```
By directly returning the response object based on conditions, the code becomes clearer and easier to understand.


<!-- This is an auto-generated comment by PullProofAI -->